.github/workflows/opam.yml

```diff
@@ -17,28 +17,24 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        ocaml-compiler: [4.12.1, 4.13.1, 4.14.1]
+        ocaml-compiler: [4.13.1, 4.14.1, 5.3.0]
 
     steps:
       - name: Checkout project
-        uses: actions/checkout@v3.5.3
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
           submodules: "recursive"
 
       - name: Setup OCaml ${{ matrix.ocaml-compiler }}
-        uses: ocaml/setup-ocaml@v2
-        if: ${{ startsWith(matrix.os, 'windows-') }}
-        with:
-          ocaml-compiler: ${{ matrix.ocaml-compiler }}
-          opam-repositories: |
-            opam-repository-mingw: https://github.com/ocaml-opam/opam-repository-mingw.git#sunset
-            default: https://github.com/ocaml/opam-repository.git
-
-      - name: Setup OCaml ${{ matrix.ocaml-compiler }}
-        uses: ocaml/setup-ocaml@v2
-        if: ${{ !startsWith(matrix.os, 'windows-') }}
+        uses: ocaml/setup-ocaml@c2e6bb92370612b89f302c3aaefa1da45ee2d702 # v3.2.15
         with:
           ocaml-compiler: ${{ matrix.ocaml-compiler }}
+      
+      - name: Override C Compiler Flags For Windows
```

Comment:
We should probably add a comment that this is needed until libbinaryen 121 is a dependency.

Reply:
Made that change.